From: NIIBE Yutaka <gniibe@fsij.org>
Date: Fri, 27 Aug 2021 01:17:19 +0000 (+0900)
Subject: gpg: Avoid use of COMMON section for all cases.
X-Git-Url: http://git.gnupg.org/cgi-bin/gitweb.cgi?p=gnupg.git;a=commitdiff_plain;h=9ad423d7218c25578dbbc1166dd8dc2b3564ca98;hp=a831f80e044a72e7db4e5e1855a5b97d85d2f90c

gpg: Avoid use of COMMON section for all cases.

* include/cipher.h (EXTERN_UNLESS_MAIN_MODULE): Remove limitation for
__riscos__.
* include/iobuf.h: Likewise.
* include/memory.h: Likewise.
* include/mpi.h: Likewise.

--

GnuPG-bug-id: 5215
Signed-off-by: NIIBE Yutaka <gniibe@fsij.org>
---

diff --git a/include/cipher.h b/include/cipher.h
index dd4af18cb..6ef6e6829 100644
--- a/include/cipher.h
+++ b/include/cipher.h
@@ -115,7 +115,7 @@ struct gcry_md_context {
 typedef struct gcry_md_context *MD_HANDLE;
 
 #ifndef EXTERN_UNLESS_MAIN_MODULE
-#if defined (__riscos__) && !defined (INCLUDED_BY_MAIN_MODULE)
+#if !defined (INCLUDED_BY_MAIN_MODULE)
 #define EXTERN_UNLESS_MAIN_MODULE extern
 #else
 #define EXTERN_UNLESS_MAIN_MODULE
diff --git a/include/iobuf.h b/include/iobuf.h
index 030f8c8e9..dfa96b286 100644
--- a/include/iobuf.h
+++ b/include/iobuf.h
@@ -70,7 +70,7 @@ struct iobuf_struct {
 };
 
 #ifndef EXTERN_UNLESS_MAIN_MODULE
-#if defined (__riscos__) && !defined (INCLUDED_BY_MAIN_MODULE)
+#if !defined (INCLUDED_BY_MAIN_MODULE)
 #define EXTERN_UNLESS_MAIN_MODULE extern
 #else
 #define EXTERN_UNLESS_MAIN_MODULE 
diff --git a/include/memory.h b/include/memory.h
index d414a9b2e..6698337e3 100644
--- a/include/memory.h
+++ b/include/memory.h
@@ -91,7 +91,7 @@ unsigned secmem_get_flags(void);
 #define DBG_MEMSTAT   memory_stat_debug_mode
 
 #ifndef EXTERN_UNLESS_MAIN_MODULE
-#if defined (__riscos__) && !defined (INCLUDED_BY_MAIN_MODULE)
+#if !defined (INCLUDED_BY_MAIN_MODULE)
 #define EXTERN_UNLESS_MAIN_MODULE extern
 #else
 #define EXTERN_UNLESS_MAIN_MODULE 
diff --git a/include/mpi.h b/include/mpi.h
index a4c16f5af..7a45ff805 100644
--- a/include/mpi.h
+++ b/include/mpi.h
@@ -36,7 +36,7 @@
 #include "memory.h"
 
 #ifndef EXTERN_UNLESS_MAIN_MODULE
-#if defined (__riscos__) && !defined (INCLUDED_BY_MAIN_MODULE)
+#if !defined (INCLUDED_BY_MAIN_MODULE)
 #define EXTERN_UNLESS_MAIN_MODULE extern
 #else
 #define EXTERN_UNLESS_MAIN_MODULE 
